contracts/cw-proposal-single/src/contract.rs

```diff
@@ -711,9 +713,90 @@ pub fn query_info(deps: Deps) -> StdResult<Binary> {
 }
 
 #[cfg_attr(not(feature = "library"), entry_point)]
-pub fn migrate(_deps: DepsMut, _env: Env, _msg: MigrateMsg) -> Result<Response, ContractError> {
-    // Don't do any state migrations.
-    Ok(Response::default())
+pub fn migrate(deps: DepsMut, env: Env, msg: MigrateMsg) -> Result<Response, ContractError> {
+    #[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug)]
+    struct BetaProposal {
+        pub title: String,
+        pub description: String,
+        /// The address that created this proposal.
+        pub proposer: Addr,
+        /// The block height at which this proposal was created. Voting
+        /// power queries should query for voting power at this block
+        /// height.
+        pub start_height: u64,
+        /// The minimum amount of time this proposal must remain open for
+        /// voting. The proposal may not pass unless this is expired or
+        /// None.
+        pub min_voting_period: Option<Expiration>,
+        /// The the time at which this proposal will expire and close for
+        /// additional votes.
+        pub expiration: Expiration,
+        /// The threshold at which this proposal will pass.
+        pub threshold: Threshold,
+        /// The total amount of voting power at the time of this
+        /// proposal's creation.
+        pub total_power: Uint128,
+        /// The messages that will be executed should this proposal pass.
+        pub msgs: Vec<CosmosMsg<Empty>>,
+
+        pub status: Status,
+        pub votes: Votes,
+        pub allow_revoting: bool,
+
+        /// Information about the deposit that was sent as part of this
+        /// proposal. None if no deposit.
+        pub deposit_info: Option<CheckedDepositInfo>,
```

Comment:
How would you feel about removing the documentation comments here? Might be nice to just add one at the top describing what these are / linking to the commit where the proposal state was this.

Reply:
+1

Reply:
agreed